WcaOnRails/app/webpacker/lib/utils/competition-table.js

```diff
@@ -0,0 +1,19 @@
+export default function calculateDayDifference(startDateString, endDateString, mode) {
+  const dateToday = new Date();
+  const startDate = new Date(startDateString);
+  const endDate = new Date(endDateString);
+  const msInADay = 1000 * 3600 * 24;
+
+  if (mode === 'future') {
+    const msDifference = startDate.getTime() - dateToday.getTime();
+    const dayDifference = Math.ceil(msDifference / msInADay);
```

Comment:
If this code works on it's own, I wouldn't recommend an extra dependency here. 
If we need to start doing timezone related stuff, I can see how we can benefit greatly from someone else doing the work however.

Recognizing that `startDateString` is likely a data with format 'yyyy-mm-dd', I already anticipate we'll get weird timezone stuff.

Reply:
The parameters are indeed in the format 'yyyy-mm-dd' as they are used for `comp.start_date` and `comp.end_date`, but currently I don't think there are timezone issues with competition dates as of now...

Is `luxon` already part of our dependencies though? I heard there were mentions of it already being used elsewhere.

Reply:
The common date issues is when you call `new Date()` on a particular date, the default time gets set to 00:00.00 in UTC, say you're in the pacific timezone, that date will be converted to 2pm in the evening the day before. When you calculate the days till, it'll be 1 less than expected.

Reply:
Ah yes this does happen. If I remember correctly the official competitions page is also off by a little bit when I checked if it was consistent with my local server so I thought it behaved fine and everything was supposed to be in UTC. I might look into `luxon` then.

Reply:
> The common date issues is when you call new Date() on a particular date, the default time gets set to 00:00.00 in UTC

Doesn't it get set to midnight in your local time zone?

I'm pretty sure Gregor was already using luxon in the codebase somewhere (venue/schedule editor?) so I'd recommend it.

Reply:
switched everything to luxon!

Reply:
I'm not sure if we should switch to using UTC or continue with using local timezones in luxon

One reason is our rails codebase can't deal with timezones. Attributes like `comp.inProgress` and `comp.isProbablyOver` are calculated based on UTC, so there's a consistency issue. In a scenario where UTC crosses midnight but when a comp's end date is still today in the local timezone, the current PR would say that they're not in progress, because they were categorized using these rails attributes, whereas the status icon calculated with luxon shows that they 'Ended 0 days ago'. To address this maybe we should determine inProgress using luxon entirely without any ruby, or switch to UTC and keep it consistent with ruby.

The other reason is using UTC to estimate when a comp is over seems more accurate than using everyone's own timezones(?), since UTC gives a better standardized sense of time, and competitions take place all over the world. 

I also found a whole discussion about `comp.is_probably_over?` here, might give some more insights: https://github.com/thewca/worldcubeassociation.org/pull/1206/files#r98485399